### PR TITLE
Feature/channel pool validation

### PR DESCRIPTION
**Pull request by itayx**
Currently the DCS's set_channel_scan_pool uses a static list of supported channels for verifications to see if the channels provided in the channel pool are supported.

The DB has a list known as hostap_supported_channels, this is filled with channels provided by the basic radio capabilities

The DCS's set_channel_scan_pool need to validate it's channel pool against this supported channels list

### CODE REVIEW

File: controller/src/beerocks/master/db/db.cpp
Comment (kantera800) [inline controller/src/beerocks/master/db/db.cpp]: please use separate method instead  `is_channel_scan_pool_supported()` 
also, no need to scan all pool , just  return false for the first channel that is not supported

Comment (itayx) [reply controller/src/beerocks/master/db/db.cpp]: added `is_channel_scan_pool_supported()`

File: controller/src/beerocks/master/db/db.cpp
Comment (CoralMalachi) [inline controller/src/beerocks/master/db/db.cpp]:
```suggestion
    for (const auto &channel : channel_pool) {
```

As far as I understand, you are only reading the `channel` value and you have no attempt to modify it, so it preferred to make it a read-only value.

**Review by CoralMalachi — APPROVED** [review]

File: controller/src/beerocks/master/db/db.cpp
Comment (itayx) [reply controller/src/beerocks/master/db/db.cpp]: Changed

File: controller/src/beerocks/master/db/db.cpp
Comment (rmelotte) [inline controller/src/beerocks/master/db/db.cpp:2305]: To me it's a bit weird that a function called `is_channel_scan_pool_supported` logs something (especially an error) if a channel is not supported. Maybe it would be best to leave that to the caller (or to change the level to INFO or DEBUG).

Aren't there any case where you would want to call `is_channel_scan_pool_supported` without logging an error?

(nitpick)

File: controller/src/beerocks/master/db/db.h
Comment (rmelotte) [inline controller/src/beerocks/master/db/db.h:521]: To me the function name is pretty explicit about what is being done already, but the documentation is so vague that it doesn't bring anything useful. 
What kind of validation does it do?

For example:

```suggestion
    /**
     * @brief Validate the channel scan pool.
     * 
     * A pool is considered valid if all the channels it contains are supported by the radio.
     * 
     * @param mac:          MAC address of the radio
     * @param channel_pool: Channel pool to validate
     * @return true if channel_pool is valid, false otherwise.
```

(nitpick)

**Review by rmelotte — APPROVED** [review]
Only nitpicks :+1:

File: controller/src/beerocks/master/db/db.cpp
Comment (itayx) [reply controller/src/beerocks/master/db/db.cpp:2305]: I completely understand
And I think I'll change it to `INFO`
But I still want the log, that way there is a log of which sent channel isn't supported
Since the function doesn't return the unsupported channel, the only way to know which is via the log

**Review by kantera800 — APPROVED** [review]

**Comment by kantera800** [comment]
approved , but don't merge it yet.
**Wait for the NO_IR issue to be solved.**

File: controller/src/beerocks/master/db/db.cpp
Comment (tomereli) [inline controller/src/beerocks/master/db/db.cpp]: This should be a separate commit (nitpick)

**Review by tomereli — APPROVED** [review]

File: controller/src/beerocks/master/db/db.cpp
Comment (itayx) [reply controller/src/beerocks/master/db/db.cpp]: This was moved to #1176 
I will remove it from here once that PR is merged

File: controller/src/beerocks/master/db/db.h
Comment (itayx) [reply controller/src/beerocks/master/db/db.h:521]: Updated the doxygen documentation

**Pull request merged**
